### ISSUE TITLE
Container and Volume sweeping relies on `gdn` and `baggageclaim` properly responding on time

### ISSUE THREAD

**Issue by cirocosta** [concourse/concourse#4467]
## Bug Report

I was looking at the container sweeping logic, and it seems to me that if a
single call to volume sweeping takes forever to respond, no sweeping at all
will take place for the next ticks (as next ticks will not occur).

e.g., the runner is essentially something that runs on every tick if never
whenever unblocked:

```go
func (sweeper *containerSweeper) Run(...) {

    for {
        select {
        case <-timer.C:
            // cc: I assume this might take forever
            //
            sweeper.sweep(sweeper.logger.Session("tick"))

        case sig := <-signals:
            sweeper.logger.Info("sweep-cancelled-by-signal", lager.Data{"signal": sig})
            return nil
        }
    }
}
```

(see [containerSweeper.Run](https://github.com/concourse/concourse/blob/ac5f6b673ff6ab9cc45b4e315c81bddba1d13352/worker/container_sweeper.go#L40-L55))

looking at `sweeper.sweep`:

```go
func (sweeper *containerSweeper) sweep(logger lager.Logger) {
    ctx := lagerctx.NewContext(context.Background(), logger)

    // cc: grab all handles ... etc
    //
    containers, err := sweeper.gardenClient.Containers(garden.Properties{})

    // cc: grab the handles to destroy
    //
    containerHandles, err := sweeper.tsaClient.ContainersToDestroy(ctx)


    // cc: for each handle, submit to the "work queue" to delete
    //
    for _, handle := range containerHandles {
        maxInFlight <- 1
        wg.Add(1)

        go func(handle string) {
            err := sweeper.gardenClient.Destroy(handle)
            if err != nil {
                logger.WithData(lager.Data{"handle": handle}).
                    Error("failed-to-destroy-container", err)
            }

            <-maxInFlight
            wg.Done()
        }(handle)
    }

    // cc: this will wait until all finish, thus, we assume
    //     that they all eventually finish on a reasonable time
    //
    wg.Wait()
}
```

(see [containerSweeper.sweep](https://github.com/concourse/concourse/blob/ac5f6b673ff6ab9cc45b4e315c81bddba1d13352/worker/container_sweeper.go#L57-L97))

ps.: if a single deletion takes forever, the current tick's containers will all be deleted (except for that faulty one), but as we wait on the entirety of it, the next tick will never happen (my hypothesis)

## Steps to Reproduce

I didn't try reproducing this, but I can see that it'd be quite simple to do an e2e verification of this by modifying Houdini to not properly do the deletion.

## Expected Results

Not prevent further ticks in the case of a deletion being stuck

## Actual Results

Gets stuck if a deletion never returns (assumption). 

## Additional Context

Note.: we've recently made the web nodes more resilient by introducing broader
use of context cancellation (see [resource checking should be resilient to a 'stuck' worker](https://github.com/concourse/concourse/issues/3928)), but it seems like we didn't do the same for the worker side.

## Version Info

* Concourse version: 5.5.1
* Deployment type (BOSH/Docker/binary): any?

**Comment by kcmannem** [comment]
This is probably important, I'd want a fix for 5.6 if we can get it in.

**Comment by kcmannem** [comment]
![Screen Shot 2019-09-19 at 4 08 02 PM](https://user-images.githubusercontent.com/9054726/65277345-b5992280-daf7-11e9-9d70-3dc79ca0f92a.png)

**Comment by cirocosta** [comment]
reproducible at https://github.com/concourse/concourse/issues/4477

**Comment by pivotal-bin-ju** [comment]
To solve the issue happened on `wings`, the first time I tried to restart `gdn server` with this command:
```
kill -9 $GDN_SERVER_PID && gdn server $GDN-PARAMETERS
```
The containers drop down from 256 to 71. (Not sure the exact concourse version)

But the 2nd time (concourse 5.5.0 on wings) I restarted the `gdn server` on another vm, it didn't work. And I found in the log it was saying connection to port `7777` is failed. 
Concourse did not find `gdn server` restarted it's own gdn again on port `7777`?
Later when I looked at directory /var/vcap/data/worker/depot, the number of folders doesn't match the number that fly returns.

Is that possible the `gdn server` is out of sync to concourse db?
